src/Nethermind/Nethermind.Evm/CodeAnalysis/CodeInfo.cs

```diff
@@ -2,23 +2,56 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
-using System.Collections;
-using System.Reflection.PortableExecutable;
-using System.Threading;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Evm.EOF;
 using Nethermind.Evm.Precompiles;
 
 namespace Nethermind.Evm.CodeAnalysis
 {
-    public class CodeInfo
+
+    public interface ICodeInfo
+    {
+        byte[] MachineCode { get; }
+        IPrecompile? Precompile { get; }
+        bool IsPrecompile => Precompile is not null;
+        bool IsEof => false;
+        byte Version => 0;
+        ReadOnlySpan<byte> TypeSection => Span<byte>.Empty;
+        ReadOnlySpan<byte> CodeSection => MachineCode;
+        ReadOnlySpan<byte> DataSection => Span<byte>.Empty;
+        bool ValidateJump(int destination, bool isSubroutine);
+    }
+
+    public class EofCodeInfo : CodeInfo
```

Comment:
I couldn't find any use place of this class, which means we never instantiate `EofCodeInfo` and that is a problem

Reply:
Yes this is great.